cmd/containerd-shim-runc-v2/task/service.go

```diff
@@ -580,8 +580,14 @@ func (s *service) Wait(ctx context.Context, r *taskAPI.WaitRequest) (*taskAPI.Wa
 	if err != nil {
 		return nil, errgrpc.ToGRPC(err)
 	}
-	p.Wait()
 
+	p.Wait(ctx)
+	select {
+	case <-ctx.Done():
+		// connection closed, discard the result
+		return nil, nil
```

Comment:
Should this be `return nil, ctx.Err()` instead?

```go
// If Done is closed, Err returns a non-nil error explaining why
```

Reply:
The previous containerd had exited, and returning err or nil here has no effect because no one will handle this err anymore.

Because there is no one to handle it, so tend to return nil directly.

Reply:
make sense, even if connection had closed, should also return context message